### ISSUE TITLE
Keras 'ModelCheckpoint' - does it accurately load the saved model! 

### ISSUE THREAD

**Issue by akashshingha850** [keras-team/keras#13720]
We know that, `ModelCheckpoint` is used to save the model weight. So, we can assume that the loaded weight model should give the last performance output. I'm training a model with 100 epoch and used `ModelCheckpoint` to save the model weight for low validation loss. 

However, from epoch 1 to 10 the loss decreased properly, and there is a good sign to get a better model. But unfortunately, after epoch 10, power failure happened. And so we reload the model with the previously saved weight, and model is still good by its performance but the issue is the **flow of improving** of the model is not exactly same as before. It fluctuates a lot. Is it normal? Should I wait for a few more epoch? I'm reporting for first 3-6 epoch.

```python
model = [define the model]
model.load_weights(..)
model.compile(...)
model.train(...)
```

**Comment by Saduf2019** [comment]
@akashshingha850 
Moving this issue to closed status as there has been no activity, in case you still face the error please create a new issue.
